runtime/runtime_plugins.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2018 The OPA Authors.  All rights reserved.
+// Use of this source code is governed by an Apache2
+// license that can be found in the LICENSE file.
+
+// +build linux,cgo darwin,cgo
+
+package runtime
+
+// Contains parts of the runtime package that use the plugin package
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"plugin"
+
+	"reflect"
+)
+
+// RegisterSharedObjectsFromDir registers all .builtin.so and .plugin.so files recursively stored in dir into OPA.
+func RegisterSharedObjectsFromDir(dir string) error {
```

Comment:
In the spirit of minimizing the number of exported functions, please move this into the cmd package and make it unexported. That way we don't have to worry about supporting it as much in the future.

Reply:
If it is unexported, it cannot be tested from a separate package. This is necessary because `go test` reloads the go runtime package and so the hashes differ: https://github.com/golang/go/issues/17928

Reply:
Ah. We can move it under the top-level internal directory then. Create a package like `opa/internal/sharedobjects` and move this functionality there.